packages/back-end/src/init/mongo.ts

```diff
@@ -5,18 +5,19 @@ import { logger } from "../util/logger";
 
 mongoose.Promise = bluebird;
 
-export default async () => {
+export default async (): Promise<void> => {
   // Connect to MongoDB
   try {
     let uri = MONGODB_URI;
     if (process.env.NODE_ENV === "test") {
       uri = process.env.MONGO_URL || "";
     }
 
-    return await mongoose.connect(uri, {
-      useNewUrlParser: true,
-      useCreateIndex: true,
-      useUnifiedTopology: true,
+    // in Mongoose 7.x, connect will no longer return a Mongoose client
+    await mongoose.connect(uri, {
+      bufferCommands: false,
+      autoCreate: true,
+      autoIndex: true,
```

Comment:
The 3 options are deprecated and are supposedly enabled by default. 

I am not entirely sure if `autoCreate`, `autoIndex` and `bufferCommands: false` are required but I had initially thought we were having issues with server startup due to the buffer timeouts Mongoose adds by default.

Mongoose 7 does not return the Mongoose client, and since we aren't using it, I've decided to explicitly return `Promise<void>` and not return the client that 6.x returns since it'll be gone once we're ready to migrate to 7 (once Agenda upgrades their mongodb version as mentioned in the PR description).

Let me know if we should change anything here.